.github/workflows/deploy-github-pages.yaml

```diff
@@ -31,28 +31,39 @@ jobs:
     # Specify runner + deployment step
     runs-on: ubuntu-latest
     steps:
+      # Check out the branch
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+      # Fix "Corepack must currently be enabled by running corepack enable in your terminal."
+      - name: Enable Corepack
+        run: corepack enable
+      # Setup Node.js with yarn support
       - name: Setup Node
         uses: actions/setup-node@v4
         with:
-          node-version: lts # Eg.: 12.x, 10.15.1, >=10.15.0, lts, latest
+          node-version: 'lts/*' # Ensures the latest LTS version is used
           cache: yarn # npm / pnpm / yarn
+      # Configure GitHub Pages
       - name: Setup Pages
         uses: actions/configure-pages@v5
+      # Install javascript dependencies with yarn
       - name: Install dependencies
         run: yarn install # or pnpm install / yarn install / bun install
+      # Build the website
       - name: Build with VitePress
         run: yarn run docs:build # or pnpm docs:build / yarn docs:build / bun run docs:build
+      # Upload the artefact to GitHub Pages
       - name: Upload artifact
-        uses: actions/upload-pages-artifact@v3
+        uses: actions/upload-pages-artifact@v4
```

Comment:
please use `7b1f4a764d45c48632c6b24a0339c27f5614fb0b` instead of `v4` to pin to commits